### DISCUSSION TITLE
Expand a specific child node in Qtree with parameter

### DISCUSSION THREAD

**Discussion by Code-Case** [quasarframework/quasar#9169] [General - Components / Directives / etc] [answered]
Hi,

is it possible to open a specific node in my qTree directly with a parameter like an id?
I would do something like http://url.com/2 to open the second node by id 2 in my tree if I enter the url.

thank you

**Comment by metalsadman** [comment]
setExpanded in methods https://quasar.dev/vue-components/tree#qtree-api

**Reply by Code-Case** [reply]
made my day mate, thanks

@metalsadman Ive tested that and it seems that I only can expand parent nodes but not their child nodes with the setexpand method?

**Comment by hawkeye64** [comment]
Maybe you can get some ideas from the code I added to the Electron File Explorer to recursively open parents until the child is found: https://github.com/hawkeye64/electron-quasar-file-explorer/blob/master/src/components/folderTree.vue#L82

**Reply by Code-Case** [reply]
Thats looks good, I will try to integrate it and post some feedback, thank you

**Reply by Code-Case** [reply]
@hawkeye64 thanks again for the example, one thing I didnt understand is. how do we get the value for 
Ive added a playground with my messi try :)

https://codepen.io/someOne11/pen/poeoBvO

**Reply by hawkeye64** [reply]
Well, that is an issue. If you don't know what to open, how can you open it? At least with the Electron File Explorer, there is a one-to-one relationship between files and the tree nodes.

**Reply by Code-Case** [reply]
Well youre right, its bad if I not know what I want to open :D

Ive reworked the code and getting now the number of children via console.log but cant expand them.
Ive also tryed to convert the objects into arrays and to split the array then like in your example because I have objects for my tree and its children.
Ive added all the fancy stuff you use like selected.sync and lazyloading, what did I miss?

https://codepen.io/someOne11/pen/rNyayQN

Thanks for all input

**Reply by hawkeye64** [reply]
@Code-Case You missed this:
![image](https://user-images.githubusercontent.com/10262924/117475619-8415da80-af19-11eb-8bb8-b2c3c02185e0.png)
After the exception, the js won't run...

**Reply by Code-Case** [reply]
Ive removed the lazyloading, thanks

**Reply by hawkeye64** [reply]
@Code-Case I am following up on it, but for an Electron app, you use "path.sep" to determine the separator linux/mac vs windows. In your case, it's undefined, so when you add it to the child, it's becoming "[object Object]undefined".

**Comment by hawkeye64** [comment, marked as answer]
@Code-Case This is what I came up with: https://codepen.io/Hawkeye64/pen/zYZxLbM

**Reply by Code-Case** [reply]
thanks thats very nice, never build an electron app so confusing about path.sep is now cleared for me.

best

**Reply by Code-Case** [reply]
@hawkeye64 Ive tryed rewrote your code a bit to just use any node key and toggle the parents and select the Item with given ID:

https://codepen.io/someOne11/pen/BaWdbdM

It seems to be working to select the child but its not toggeling the parents above?

Maybe your magic mind could help me out?

Its because I just have a ID no path to expand the tree to given element from url.

**Reply by hawkeye64** [reply]
I think also toggling the parents has something to do with the tick strategy

**Reply by Code-Case** [reply]
I use now something like this : https://codepen.io/donnyj42/pen/QWpOexj

Works like I needed it till the third level
You have to know the id of the root node
but I think thats stuff all know from the tree data

**Reply by metalsadman** [reply]
Research about recursive function, you should be able to dig infinitely into deeply nested tree like data structure using it. Tons of resource about it out there, it's generic so don't have to be quasar specific.

**Reply by Code-Case** [reply]
I will, thanks
